homeassistant/helpers/trigger.py

```diff
@@ -621,3 +625,103 @@ async def async_get_all_descriptions(
 
     hass.data[TRIGGER_DESCRIPTION_CACHE] = new_descriptions_cache
     return new_descriptions_cache
+
+
+class TargetSelectorStateChangeTracker:
```

Comment:
Maybe this class would fit better in `event.py` together with the other state change event listeners?

Reply:
Moved to `target.py` as discussed in DM.